File: 2026/04-April/28-Tuesday/solution.py
class Solution:
    def minOperations(self, grid, x):
        nums = [num for row in grid for num in row]
        nums.sort()

        base = nums[0] % x
        for num in nums:
            if num % x != base:
                return -1 

        median = nums[len(nums) // 2]
        return sum(abs(num - median) // x for num in nums)

File: 2026/04-April/28-Tuesday/solution.java
import java.util.*;

class Solution {
    public int minOperations(int[][] grid, int x) {
        List<Integer> nums = new ArrayList<>();

        for (int[] row : grid) {
            for (int num : row) {
                nums.add(num);
            }
        }

        Collections.sort(nums);

        int base = nums.get(0) % x;
        for (int num : nums) {
            if (num % x != base) return -1;
        }

        int median = nums.get(nums.size() / 2);
        int ops = 0;

        for (int num : nums) {
            ops += Math.abs(num - median) / x;
        }

        return ops;
    }
}
